app/models/proposed_organisation.rb

```diff
@@ -1,7 +1,8 @@
 class ProposedOrganisation < BaseOrganisation
   has_many :users, :foreign_key => 'organisation_id'
-  def accept_proposal
-    org = becomes!(Organisation)
+  
+  def accept_proposal(rollback = false)
+    org = rollback ? becomes!(ProposedOrganisation) : becomes!(Organisation)
```

Comment:
I understand that there is change from `Organisation` to `ProposedOrganisation`  it will be better if you add a method to `Organisation` instead of adding a control flag (it's called control coupling )